src/ucm/cuda/cudamem.c

```diff
@@ -434,7 +461,7 @@ static int ucm_cudamem_scan_regions_cb(void *arg, void *addr, size_t length,
 
     event.mem_type.address  = addr;
     event.mem_type.size     = length;
-    event.mem_type.mem_type = UCS_MEMORY_TYPE_LAST; /* unknown memory type */
+    event.mem_type.mem_type = ucm_cuda_get_mem_type(addr);
```

Comment:
@dmitrygx we can't do this.  the whole mapped region from cuda driver can be used for any purpose. the might contain HOST, CUDA or MANAGED..etc. we are setting UCS_MEMORY_TYPE_LAST here so that it will actually detect the buffer used for communication with UCTs ( https://github.com/openucx/ucx/blob/532e172dabb4b058e641c3ba161fca7ea561501b/src/ucp/core/ucp_context.h#L419) during runtime.

Reply:
are we sure cudaPointerGetAttributes retuns not success for such regions

Reply:
@bureddy yes, you're right this is not needed, it always return `LAST`

Reply:
> are we sure cudaPointerGetAttributes retuns not success for such regions

yes, it returns not OK here, but I agree that we need to mark such regions as `UCS_MEMORY_TYPE_LAST` - will return it back

Reply:
@bureddy fixed